Terminal.Gui/Application.cs

```diff
@@ -317,6 +317,8 @@ public static RunState Begin (Toplevel Toplevel)
 			} else if (Top != null && Toplevel != Top && _topLevels.Contains (Top)) {
 				Top.OnLeave (Toplevel);
 			}
+			// BUGBUG: We should not depend on `Id` internally. 
+			// BUGBUG: It is super unclear what this code does anyway.
 			if (string.IsNullOrEmpty (Toplevel.Id)) {
 				int count = 1;
 				string id = (_topLevels.Count + count).ToString ();
```

Comment:
Well I think the id start from 1 forward. If _topLevels.Count == 0 + 1 will return 1. Another option is:

`string id = (_topLevels.Count + 1).ToString ();`